File: nivel1/problema_de_las_divisiones_nivel1/dividiendo.py
def dividiendo(num,veces):
    for i in range(veces):
        num = int(num/2)
        print(num)
    return num
numero = 1
while numero != 0:
    numero = input("Ingrese un numero: ")

    while not numero.isdigit():
        numero = input("Ingreselo nuevamente: ")

    numero = int(numero)
    if numero == 0: break
    
    veces = input("Ingrese las veces que quiere dividir: ")
    while not veces.isdigit():
        veces = input("Ingrese nuevamente las veces: ")
    veces = int(veces)
    
    print(dividiendo(numero,veces))

File: jurisdiccional/terminados/problema_de_las_divisiones_nivel1/dividiendo.cpp
#include<iostream>
#include<cmath>
#include<limits>
using namespace std;
int pedirNumero(string texto)
{
        int numero; 
        cout << texto ; cin >> numero;    
        while(true){
            if (cin.fail())
            {
                
                cin.clear();
                cin.ignore(numeric_limits<streamsize>::max(), '\n');
                cout << "Vuelve a ingresar el numero: "; cin >> numero;  
            }
            else break;  
        }
        return numero;
}

void dividiendo(int num, int  veces)
{
    float numero = num;
    for(int i = 0; i < veces; i++)
    {
        numero = numero / 2;
        numero = trunc(numero);
        cout << numero << endl;
    }
}

int main()
{
    int numero = 1;
    while (numero)
    {   
        numero = pedirNumero("Ingrese un numero: ");
        if (!numero) break;
        int veces = pedirNumero("Ingrese la cantidad de veces que quiere dividir: ");
        dividiendo(numero,veces);
    }
    
}





